[BUG] Healthcheck error with 1.7.0-rc1

### Description

After upgrade to 1.7.0-rc1 the docker healthcheck is no longer working

    healthcheck:
      test: ["CMD", "curl", "-f", "http://localhost:8000"]

I assume this is connected to https://github.com/paperless-ngx/paperless-ngx/pull/674

### Expected behavior

Healthcheck working, to see that container is alive

### Steps to reproduce

Update to 1.7.0-rc1 using healthcheck as documented in https://github.com/paperless-ngx/paperless-ngx/blob/main/docker/compose/docker-compose.portainer.yml

    healthcheck:
      test: ["CMD", "curl", "-f", "http://localhost:8000"]
      interval: 30s
      timeout: 10s
      retries: 5

### Webserver logs

```bash
[2022-04-12 10:36:56,641] [ERROR] [django.security.DisallowedHost] Invalid HTTP_HOST header: 'localhost:8000'. You may need to add 'localhost' to ALLOWED_HOSTS.
[2022-04-12 10:36:56,673] [WARNING] [django.request] Bad Request: /
```


### Screenshots

_No response_

### Paperless-ngx version

1.7.0-rc1

### Host OS

Synology DSM 7.1

### Installation method

Docker

### Browser

Edge

### Configuration changes

No changes in docker compose regarding healthcheck, using standard healthcheck configuration as documented

### Other

_No response_

--- a/docs/configuration.rst
+++ b/docs/configuration.rst
@@ -170,6 +170,9 @@ PAPERLESS_ALLOWED_HOSTS=<comma-separated-list>
 
     Can also be set using PAPERLESS_URL (see above).
 
+    If manually set, please remember to include "localhost". Otherwise docker
+    healthcheck will fail.
+
     Defaults to "*", which is all hosts.
 
 PAPERLESS_CORS_ALLOWED_HOSTS=<comma-separated-list>

--- a/src/paperless/settings.py
+++ b/src/paperless/settings.py
@@ -251,7 +251,8 @@ if _paperless_url:
     if _allowed_hosts:
         ALLOWED_HOSTS.append(_paperless_uri.hostname)
     else:
-        ALLOWED_HOSTS = [_paperless_uri.hostname]
+        # always allow localhost. Necessary e.g. for healthcheck in docker.
+        ALLOWED_HOSTS = [_paperless_uri.hostname] + ["localhost"]
 
 # The secret key has a default that should be fine so long as you're hosting
 # Paperless on a closed network.  However, if you're putting this anywhere

